[bugfix] fix `eayunstack init` run on fuel ERROR

blkart opened the pull request:
Ignore the node in discover status when run eayunstack init command.

redmine 7588

Signed-off-by: blkart blkart.org@gmail.com

huntxu commented on eayunstack_tools/init/__init__.py:
```diff
@@ -53,7 +53,9 @@ def init_node_list_file():
-        if len(node['roles']) > 1:
+        if len(node['roles']) == 0:
+            continue
+        elif len(node['roles']) > 1:
```
和这个改动不太相关，但这里整个if-elif-else其实就是一行`roles=','.join(node['roles'])`吧，然后可以直接判断`roles`还是为空跳过这个节点，或者照样用上面第一个`if`判断列表长度也没问题。只是这里下面逻辑有点多余，没必要按照长度判断去选择合并出roles的方式，既然要改这里我就建议一并改了：

```
# 从55行开始到65行
roles = ','.join(node['roles'])
if not roles:
    continue
```

blkart committed:
[bugfix] fix `eayunstack init` run on fuel ERROR

Ignore the node in discover status when run eayunstack init command.

redmine 7588

Signed-off-by: blkart <blkart.org@gmail.com>

huntxu commented:
难得有我看懂的 :P
